src/libraries/Stack.sol

```diff
@@ -0,0 +1,50 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+
+contract Stack {
+    
+    var[] collection;
+    uint head;
+    address owner;
+    bool initialized;
+
+    function any() {
+         if (initialized) {
+             require(msg.sender == this.owner || msg.sender == this);
+         }
+    }
+
+    function initialize(address owner) returns (uint) {
+         require(!this.initialized);
+         this.initialized = true;
+         this.owner = owner;
+         return (uint(1));
+    }
+
+    function push(var item) returns (uint) {
+         uint index = this.head + 1;
+         this.head = index;
+         this.collection[index] = item;
+         return (uint(1));
+    }    
+
+    function pop() returns (uint) {
```

Comment:
Should this be returning a uint?  Isn't removedValue set to a pointer in this case?  In push(), this.collection[index] looks like it's getting set to a pointer.